src/Models/LoginSession.php

```diff
@@ -305,4 +306,33 @@ public static function getSessionLifetime(): int
 
         return LoginSession::config()->get('default_session_lifetime');
     }
+
+    /**
+     * Update LastAccessed date and IP address
+     *
+     * @param HTTPRequest $request
+     * @return void
+     */
+    public function updateLastAccessed(HTTPRequest $request = null)
+    {
+        $this->LastAccessed = DBDatetime::now()->Rfc2822();
+        $this->IPAddress = $this->getIpFromRequest($request);
+        $this->write();
+    }
+
+    /**
+     * @param HTTPRequest|null $request
+     * @return string
+     */
+    private function getIpFromRequest(HTTPRequest $request = null): string
+    {
+        if (!$request) {
+            return '';
+        }
+        $ip = $request->getIP();
+        if (static::config()->anonymize_ip) {
```

Comment:
```suggestion
        if (static::config()->get('anonymize_ip')) {
```
We don't use the config-as-property syntax for fetching configuration in new code.